### DISCUSSION TITLE
AlloyDB- CREATE DATABASE can not be run in a transaction block

### DISCUSSION THREAD

**Discussion by ff-sdesai** [sqlalchemy/sqlalchemy#10592] [Usage Questions] [answered]
### Describe the bug

Since AlloyDB does not support creating a database using terraform yet, I am trying to create it using Alembic and sqlAlchemy. I am using postgresql+psycopg2 as the driver.My code looks similar to the only answer on [here](https://stackoverflow.com/questions/53847829/postgres-pg8000-create-database).

```
from sqlalchemy import create_engine
dburl = "postgresql+psycopg2://user:pswd@myip:5432/postgres/"
engine = create_engine(dburl)
conn = engine.connect()

con.rollback()  # Make sure we're not in a transaction
con.autocommit = True  # Turn on autocommit

conn.execute("CREATE DATABASE qux")
 
con.autocommit = False  # Turn autocommit back off again
```
However, I am getting an error `sqlalchemy.exc.InternalError: (psycopg2.errors.ActiveSqlTransaction) CREATE DATABASE cannot run inside a transaction block`. Is this expected when I am turning on auto-commit and rollick back transaction?

Additional metadata-

OS: Ubuntu 22.04
Psycopg version: 2.9.9
Python version: 3.10
PostgreSQL version: AlloyDB
pip version- 3

### SQLAlchemy Version in Use

2.0.22

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

AlloyDB- corresponding to PostgreSQL 14

### Python Version

3.10

### Operating system

Ubuntu 22.04

### To Reproduce

```python
from sqlalchemy import create_engine
dburl = "postgresql+psycopg2://user:pswd@myip:5432/postgres/"
engine = create_engine(dburl)
conn = engine.connect()

con.rollback()  # Make sure we're not in a transaction
con.autocommit = True  # Turn on autocommit

conn.execute("CREATE DATABASE qux")
```

### Error

 sqlalchemy.exc.InternalError: (psycopg2.errors.ActiveSqlTransaction) CREATE DATABASE cannot run inside a transaction block.

**Comment by zzzeek** [comment, marked as answer]
SQLAlchemy [Connection](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Connection) objects are not the same thing as the DBAPI connection that you would get from your psycopg2 driver.    They are facades on top which have a different API and usage patterns.

To set a `Connection` into autocommit you want to use an execution option.  For complete documentation on setting autocommit on `Connection` objects see https://docs.sqlalchemy.org/en/20/core/connections.html#understanding-the-dbapi-level-autocommit-isolation-level

**Reply by ff-sdesai** [reply]
Thanks @zzzeek . I was able to make it work using execution option
